### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by NeverovDS**
Function that returns an average value of array

### CODE REVIEW

File: FoundAverage.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package main
+
+import "fmt"
+
+func average(arr [] int) float64 {
+	 Sum  := 0
+	N := len(arr)
```
Comment (burov) [inline FoundAverage.go]: local variable name should start from lower case letter

File: FoundAverage.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package main
+
+import "fmt"
+
+func average(arr [] int) float64 {
+	 Sum  := 0
```
Comment (burov) [inline FoundAverage.go]: format issue

File: FoundAverage.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	N := len(arr)
+	for i := 0; i < len(arr) ; i++ {
+	Sum += arr[i]
```
Comment (burov) [inline FoundAverage.go]: format issue

File: FoundAverage.go (hunk 2 above)
Comment (burov) [inline FoundAverage.go]: local variable name should start from lower case letter

File: FoundAverage.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+	}
+	return float64(Sum) / float64(N)
+	}
+	func main() {
+
+	 SomeArray := [] int{4,2,3,1}
```
Comment (burov) [inline FoundAverage.go]: see comment above

File: FoundAverage.go [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import "fmt"
+
+func average(arr [] int) float64 {
+	 Sum  := 0
+	N := len(arr)
+	for i := 0; i < len(arr) ; i++ {
+	Sum += arr[i]
+	}
+	return float64(Sum) / float64(N)
+	}
```
Comment (burov) [inline FoundAverage.go]: format issue

File: FoundAverage.go [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+	func main() {
+
+	 SomeArray := [] int{4,2,3,1}
+	 a := average(SomeArray)
+
+		fmt.Println(a)
```
Comment (burov) [inline FoundAverage.go]: format issue

File: FoundAverage.go (hunk 5 above)
Comment (imatsiash-ib) [reply FoundAverage.go]: What happens if len(arr) == 0 ?
Please add a check that slice length is greater than 0.

File: FoundAverage.go (hunk 1 above)
Comment (NeverovDS) [reply FoundAverage.go]: fix

File: FoundAverage.go (hunk 5 above)
Comment (NeverovDS) [reply FoundAverage.go]: fix

File: FoundAverage.go [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+func average(arr [] int) float64 {
+	sum := 0
+	n := len(arr)
+	if len(arr) != 0 {
+
+		for i := 0; i < len(arr); i++ {
+			sum += arr[i]
+		}
+	}
+	return float64(sum) / float64(n)
```
Comment (imatsiash-ib) [inline FoundAverage.go]: If `len(arr)` == 0, we will divide by zero here.

Comment (NeverovDS) [reply FoundAverage.go]: fix)

File: FoundAverage.go [hunk 8]
```diff
@@ -0,0 +1,24 @@
+package main
+
+import (
+	"fmt"
+)
+
+func average(arr [] int) float64 {
+	sum := 0
+	n := len(arr)
+	if len(arr) == 0 {fmt.Println("Error")}
```
Comment (imatsiash-ib) [inline FoundAverage.go]: I'd suggest to return 0 in case of len == 0.

Comment (NeverovDS) [reply FoundAverage.go]: done

File: FoundAverage.go [hunk 9]
```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"fmt"
+)
+
+func average(arr [] int) float64 {
+	sum := 0
+	n := len(arr)
+	if len(arr) == 0 {
```
Comment (imatsiash-ib) [inline FoundAverage.go]:
```suggestion
	n == 0 {
```

File: FoundAverage.go [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+		return 0
+	}
+	if len(arr) != 0 {
```
Comment (imatsiash-ib) [inline FoundAverage.go]:
```suggestion
	if n != 0 {
```

File: FoundAverage.go [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+
+		for i := 0; i < len(arr); i++ {
```
Comment (imatsiash-ib) [inline FoundAverage.go]:
```suggestion
		for i := 0; i < n; i++ {
```

Comment (NeverovDS) [reply FoundAverage.go]: +

File: FoundAverage.go [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+	"fmt"
+)
+
+func average(arr [] int) float64 {
+	sum := 0
+	n := len(arr)
+	if n == 0 {
+		return 0
+	}
+	if n != 0 {
```
Comment (imatsiash-ib) [inline FoundAverage.go]: This `if` is redundant here. We already returned 0 in case of n == 0.

Comment (NeverovDS) [reply FoundAverage.go]: +

**Pull request closed without merging**